### DISCUSSION TITLE
Using Pretrained model for inverse problem

### DISCUSSION THREAD

**Discussion by Rahulx084** [lululxvi/deepxde#1354] [General]
Hi, is there a way that we can use a pre-trained model from a forward problem and use it for the inverse problem?

**Comment by lululxvi** [comment]
Yes, you can save the model for futher use. See FAQ.

**Reply by Rahulx084** [reply]
I  saved a model using model_save_path which was trained for a forward problem , but when I do model_restore and use it for an inverse problem which have the same physics(just that one parameter in PDE is made variable) and architecture.It gives an error since the number of trainable parameter has increased. Is there anything that I can do to use the saved model in the inverse problem passed using model
_restore_path ??

**Reply by lululxvi** [reply]
Define those variables in advance when training the network.

**Reply by Rahulx084** [reply]
What it means by defining those variables in advance? I do this , lets say b is the parameter that I want to predict. I define it like this , b=dde.variable(1.0) . Anything else is there to do? I define it this way and still get that error. If there's anything else to do , please explain. Thanks.

**Reply by lululxvi** [reply]
It should work. There may be other issues in your code.
